tests/test_compat.py

```diff
@@ -0,0 +1,69 @@
+from django.test import TestCase
+
+from rest_framework import compat
+
+
+class CompatTests(TestCase):
+
+    def test_total_seconds(self):
+        class MockTimedelta(object):
+            days = 1
+            seconds = 1
+            microseconds = 100
+        timedelta = MockTimedelta()
+        expected = (timedelta.days * 86400.0) + float(timedelta.seconds) + (timedelta.microseconds / 1000000.0)
+        assert compat.total_seconds(timedelta) == expected
+
+    def test_get_remote_field_with_old_django_version(self):
+        class MockField(object):
+            rel = 'example_rel'
+        original_django_version = compat.django.VERSION
+        compat.django.VERSION = (1, 8)
+        assert compat.get_remote_field(MockField(), default='default_value') == 'example_rel'
+        assert compat.get_remote_field(object(), default='default_value') == 'default_value'
+        compat.django.VERSION = original_django_version
+
+    def test_get_remote_field_with_new_django_version(self):
+        class MockField(object):
+            remote_field = 'example_remote_field'
+        original_django_version = compat.django.VERSION
+        compat.django.VERSION = (1, 10)
+        assert compat.get_remote_field(MockField(), default='default_value') == 'example_remote_field'
+        assert compat.get_remote_field(object(), default='default_value') == 'default_value'
+        compat.django.VERSION = original_django_version
```

Comment:
This won't reset the Django version if this test fails, because an `AssertionError` will be raised before it gets to this point.